src/services/word.service.ts

```diff
@@ -23,47 +24,63 @@ export class WordService {
   ) {}
 
   /** Post a new word */
-  async post(postReqDto: PostWordBodyDTO): Promise<WordDomain> {
+  async post(
+    atd: AccessTokenDomain,
+    postReqDto: PostWordBodyDTO,
+  ): Promise<WordDomain> {
     if (!postReqDto.example) {
       // If no example given, Ask Chat GPT to generate one, if allowed.
       postReqDto.example = await this.termToExamplePrompt.get(postReqDto.term)
     }
 
     return WordDomain.fromMdb(
-      await WordDomain.fromPostReqDto(postReqDto)
+      await WordDomain.fromPostReqDto(atd, postReqDto)
         .toDocument(this.deprecatedWordModel)
         .save(),
     )
   }
 
   /** Get words by given query */
-  async get(query: GetWordQueryDTO): Promise<Partial<IWord>[]> {
+  async get(
+    atd: AccessTokenDomain,
+    query: GetWordQueryDTO,
+  ): Promise<Partial<IWord>[]> {
     return (
       await this.deprecatedWordModel
         .find(
-          this.getWordQueryFactory.getFilter(query),
+          this.getWordQueryFactory.getFilter(query, atd),
           this.getWordQueryFactory.getProjection(),
           this.getWordQueryFactory.getOptions(query),
         )
         .sort(this.getWordQueryFactory.toSort())
         .limit(query.limit)
         .exec()
-    ).map((props) => WordDomain.fromMdb(props).toResDTO())
+    ).map((wordRaw) => WordDomain.fromMdb(wordRaw).toResDTO())
   }
 
   /** Get word ids by given query */
-  async getWordIds(query: GetWordQueryDTO): Promise<GetWordIdsRes> {
-    const words = await this.get(query)
+  async getWordIds(
+    atd: AccessTokenDomain,
+    query: GetWordQueryDTO,
+  ): Promise<GetWordIdsRes> {
+    const words = await this.get(atd, query)
     return {
       length: words.length,
       wordIds: words.map((e) => e.id),
     }
   }
 
   /** Get word data by given id */
-  async getById(id: string): Promise<Partial<IWord>> {
-    return WordDomain.fromMdb(
-      await this.deprecatedWordModel.findById(id).exec(),
-    ).toResDTO()
+  async getById(atd: AccessTokenDomain, id: string): Promise<Partial<IWord>> {
+    // TODO: Make these one liner with factory.
+    const query = new GetWordQueryDTO()
+    query.id = id
+    query.limit = 1
+    query.userId = atd.userId
+
+    const res = await this.get(atd, query)
+    if (res.length !== 1) throw new Error('Not found!')
+
+    return res[0]
```

Comment:
Looks bad.
Should leverage the speed you get for getById of mdb.
After you get mdb the WordDomain.toRes() should have
atd that should throw unauthorized error if userId(ownerId) does not match

Reply:
At this point it requires atd, and when it fails with atd it will cause an error